### DISCUSSION TITLE
What to do when an application ID is too long?

### DISCUSSION THREAD

**Discussion by Lanerun** [microsoft/winget-cli#4263] [Q&A]
I'd like to update a specific application, without running the upgrade all command. The problem is that one application name is too long, which leads to it being cut off. Anyone have any ideas on what I can do? I tried using the partial ID, guessing that it might autocomplete it, but alas, I found no success.
![image](https://github.com/microsoft/winget-cli/assets/155983714/94759b61-5088-4453-b81c-cf7263244342)

**Comment by Trenly** [comment]
You can try using the partial Name - `winget upgrade "CrystalDiskInfo 9.2.1 Shizuku"`

**Reply by Lanerun** [reply]
> You can try using the partial Name - `winget upgrade "CrystalDiskInfo 9.2.1 Shizuku"`

@Trenly
No luck sadly (tried both powershell and command prompt) 

![image](https://github.com/microsoft/winget-cli/assets/155983714/3f1f6317-1242-47b0-9738-7b83ddf33b7b)
![image](https://github.com/microsoft/winget-cli/assets/155983714/37da581d-4132-4064-928f-2de7d6179e98)

**Reply by Trenly** [reply]
Darn, I was hoping that would work :(
There is a way to add [Autocomplete](https://github.com/microsoft/winget-cli/blob/master/doc/Completion.md) in PowerShell, which should allow you to Auto-complete the Id, but it has to be enabled manually

You can also try the `winget complete` command if you want to call just a single instance of the autocompleter -
```
winget complete --commandline="winget upgrade --id CrystalDewWorld.CrystalDiskInfo" --word="CrystalDewWorld.CrystalDiskinfo" --position=999
```

**Reply by Lanerun** [reply]
> Darn, I was hoping that would work :( There is a way to add [Autocomplete](https://github.com/microsoft/winget-cli/blob/master/doc/Completion.md) in PowerShell, which should allow you to Auto-complete the Id, but it has to be enabled manually
> 
> You can also try the `winget complete` command if you want to call just a single instance of the autocompleter -
> 
> ```
> winget complete --commandline="winget upgrade --id CrystalDewWorld.CrystalDiskInfo" --word="CrystalDewWorld.CrystalDiskinfo" --position=999
> ```
@Trenly

I see! Should I do anything before pasting that? I tried just copying it in, but it didn't seem to do much. Good to know there's a command for it at least. 

![image](https://github.com/microsoft/winget-cli/assets/155983714/1ea46974-b500-4901-ae13-1279e41c2ece)

**Reply by Trenly** [reply]
Ah, my mistake, looks like I forgot to ensure the casing was the same on CrystalDiskInfo

**Reply by Lanerun** [reply]
> Ah, my mistake, looks like I forgot to ensure the casing was the same on CrystalDiskInfo

Didn't notice that either. Works now, thanks.
